docs/docs/aleph_zero/sdk.md

```diff
@@ -257,24 +236,101 @@ const sqrtPriceLimit = calculateSqrtPriceAfterSlippage(
 )
 
 const swapResult = await invariant.swap(account, poolKey, true, amount, true, sqrtPriceLimit)
-console.log(swapResult.hash) // print transaction hash
+console.log(swapResult.hash)
 ```
 
-### Query state
+:::tip Output
+0xd9cdfddb2c783f24a481811f0f9d7037e2f7202907f092986ecd98838db2b3cb
+:::
+
+### List of Queries and Interfaces
+
+Here are some possible queries and their corresponding TypeScript interfaces:
+
+- Get Tick
+
+```typescript
+interface Tick {
+  index: bigint
+  sign: boolean
+  liquidityChange: Liquidity
+  liquidityGross: Liquidity
+  sqrtPrice: SqrtPrice
+  feeGrowthOutsideX: FeeGrowth
+  feeGrowthOutsideY: FeeGrowth
+  secondsOutside: bigint
+}
+
+const tickState: Tick = await invariant.getTick(signer, poolKey, tickIndex)
+```
+
+- Get Pool
+
+```typescript
+interface Pool {
+  liquidity: Liquidity
+  sqrtPrice: SqrtPrice
+  currentTickIndex: bigint
+  feeGrowthGlobalX: FeeGrowth
+  feeGrowthGlobalY: FeeGrowth
+  feeProtocolTokenX: TokenAmount
+  feeProtocolTokenY: TokenAmount
+  startTimestamp: bigint
+  lastTimestamp: bigint
+  feeReceiver: string
+}
+
+const poolState: Pool = await invariant.getPool(signer, TOKEN0_ADDRESS, TOKEN1_ADDRESS, feeTier)
+```
+
+- Get Pools
+
+```typescript
+const pools: Pool[] = await invariant.getPools(signer)
+```
 
 :::note Position indexing
-Remember that positions are indexed from 0. So if you create position, its id will be 0 and your next positions id will be 1.
+Remember that positions are indexed from 0. So if you create position, its index will be 0 and your next positions index will be 1.
 :::
 
+- Get Position
+
+```typescript
+interface Position {
+  poolKey: PoolKey
+  liquidity: Liquidity
+  lowerTickIndex: bigint
+  upperTickIndex: bigint
+  feeGrowthInsideX: FeeGrowth
+  feeGrowthInsideY: FeeGrowth
+  lastBlockNumber: bigint
+  tokensOwedX: TokenAmount
+  tokensOwedY: TokenAmount
+}
+
+const positionState: Position = await invariant.getPosition(signer, owner.address, positionIndex)
+```
+
+- Get Positions
+
+```typescript
+const positions: Position[] = await invariant.getPositions(
+  await invariant.getPositions(signer, owner.address)
+)
+```
+
+### Query states and Calculate Fee
+
+To query the state and calculate fees within the system, several functions are utilized. Positions, ticks, and pools are accessed to gather information about the state, and the calculateFee function is used to determine the amount of unclaimed tokens. The process is described below:
```

Comment:
"To query the state and calculate fees within the system" -> "To query the state and calculate **unclaimed** fees **belonging to the position**"